tests/unit/test_dbus_message.cpp

```diff
@@ -147,7 +208,87 @@ TEST(DBusMessage, TestStructMessage)
     TupleToDBusMessage(*msg, setVals);
     DBusMessageToTuple(*msg, getVals);
 
+    printf("%zu %s\n", std::get<2>(getVals).size(), std::get<2>(getVals)[0].c_str());
     assert(setVals == getVals);
 
     dbus_message_unref(msg);
 }
+
+TEST(DBusMessage, TestOtbrChannelQuality)
+{
+    DBusMessage *                         msg = dbus_message_new(DBUS_MESSAGE_TYPE_METHOD_RETURN);
+    tuple<otbr::DBus::otbrChannelQuality> setVals({1, 2});
+    tuple<otbr::DBus::otbrChannelQuality> getVals;
+
+    assert(msg != NULL);
+
+    TupleToDBusMessage(*msg, setVals);
+    DBusMessageToTuple(*msg, getVals);
+
+    assert(std::get<0>(setVals) == std::get<0>(getVals));
+
+    dbus_message_unref(msg);
+}
+
+TEST(DBusMessage, TestOtbrChildInfo)
+{
+    DBusMessage *                    msg = dbus_message_new(DBUS_MESSAGE_TYPE_METHOD_RETURN);
+    tuple<otbr::DBus::otbrChildInfo> setVals({1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, true, false, true, false, true});
+    tuple<otbr::DBus::otbrChildInfo> getVals;
+
+    assert(msg != NULL);
+
+    TupleToDBusMessage(*msg, setVals);
+    DBusMessageToTuple(*msg, getVals);
+
+    assert(std::get<0>(setVals) == std::get<0>(getVals));
+
+    dbus_message_unref(msg);
+}
+
+TEST(DBusMessage, TestOtbrNeighborInfo)
+{
+    DBusMessage *                       msg = dbus_message_new(DBUS_MESSAGE_TYPE_METHOD_RETURN);
+    tuple<otbr::DBus::otbrNeighborInfo> setVals({1, 2, 3, 4, 5, 6, 7, 8, 9, 10, true, false, true, false, true});
+    tuple<otbr::DBus::otbrNeighborInfo> getVals;
+
+    assert(msg != NULL);
+
+    TupleToDBusMessage(*msg, setVals);
+    DBusMessageToTuple(*msg, getVals);
+    assert(std::get<0>(setVals) == std::get<0>(getVals));
+
+    dbus_message_unref(msg);
+}
+
+TEST(DBusMessage, TestOtbrLeaderData)
+{
+    DBusMessage *                     msg = dbus_message_new(DBUS_MESSAGE_TYPE_METHOD_RETURN);
+    tuple<otbr::DBus::otbrLeaderData> setVals({1, 2, 3, 4, 5});
+    tuple<otbr::DBus::otbrLeaderData> getVals;
+
+    assert(msg != NULL);
```

Comment:
use gtest `ASSERT_*`?

Reply:
In fact it's cpputest but I'll replace the assert with the CHECK macro from the test framework.